### PR TITLE
feat(table): added striped rows

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/4564

### CODE REVIEW

**Comment by mcoker** [comment]
@mcarrano here's a POC of all of the tables with row striping. Some tables may not be able to be styled automatically via CSS since what we can do is style "every other" (odd or even) row, or every other `tbody` element's primary row, depending on how the table is constructed. This works best with simple tables with a single `tbody`. Where it doesn't work as well is compound expansion (that example is broken currently), expandable rows (though it seems to work OK), and a user composed table that might look like this:

```
table parent
  tbody
    trow
    trow
    trow
  tbody
    trow
    trow
    trow
  tbody
    trow
    trow
    trow
```

In this scenario we would need to know to style the odd rows in the first tbody, even rows in the second tbody, and odd rows again in the third tbody, which we can't do with just CSS.

For the example above and other custom/non-standard structures needing striping, we could provide different classes to stripe both odd and even rows and leave it to the user to figure out where to use the classes.

We could see if react can try and figure out which rows to apply a class to, and apply a class to each row that gets the striped background color. That's probably a lot of logic though, I'm guessing.

Or we could limit this to use in simple tables with a flat/known row structure, and have a manual striped row class if a product has a more complex table and wants to add striping manually. wdyt?

cc @mattnolting - any other ideas?

**Comment by mcarrano** [comment]
@mcoker I can see where this could get complicated in certain configurations. Maybe we do limit the usage to simple tables.

I did also want to ask about these examples for hoverable/selectable rows: https://patternfly-pr-4569.surge.sh/components/table#hoverable-and-selected It looks like there's a different hover appearance for white vs gray rows or is that my imagination?

Also, are these filled rows using BLACK-100? The look darker than the design mocks.

**Comment by mcoker** [comment]
>  It looks like there's a different hover appearance for white vs gray rows

@mcarrano nothing changed there regarding the hover behavior. When you hover over a row, a shadow appears above/below that row, so that shadow is more prominent when you hover a grey background row (since the shadow appears on the white rows above/below it) compared to hovering a white background row (since the shadow appears on grey rows above/below it, less visible since the shadow is grey)

> are these filled rows using BLACK-100?

Nope! I assumed it was background-color--200 (black-200) - updated it to be black-100.

**Comment by mcarrano** [comment]
Hmmmm... After the last update, I'm not seeing the row striping at all.

**Comment by mcoker** [comment]
@mcarrano oops, had a typo - updated.

**Comment by mcarrano** [comment]
Yep. That looks good now.

**Review by mattnolting — COMMENTED** [review]
@mcoker @mcarrano I took a stab at some creative and tricky CSS styling based on `tbody` odd/even row combinators, the result was overly complex and unreliable. 

> @mcoker I can see where this could get complicated in certain configurations. Maybe we do limit the usage to simple tables.

Seems to me that you've outlined the best and most reliable path forward above.

**Review by mcarrano — COMMENTED** [review]
@mcoker Looks good. But just a couple questions/requests.

* I think we agreed that it is OK to restrict this to simple tables. Is it possible to also support tables with simple expansion as in the design mocks?
* Can we include documentation that calls out exactly what table variations support this (and what does not)? I can see someone trying to apply this solution to a complex table and wondering why it does not work.

**Comment by mcoker** [comment]
@mcarrano yep, added a blurb under the main striped section header. I called out what is supported, implying variations not listed are not supported (which keeps us from having to maintain a list as we add variations). Let me know if that's OK or if you'd like me to be more explicit.

Expandable was supported, so added an example of that under the striped section.

**Review by mcarrano — APPROVED** [review]
Looks great @mcoker . Thanks!

File: src/patternfly/components/Table/table.scss [hunk 1]
```diff
@@ -303,6 +303,11 @@
   // Subhead
   --pf-c-table__subhead--Color: var(--pf-global--Color--200);
 
+  // Striped
+  --pf-c-table--m-striped--n-child: odd;
+  --pf-c-table--m-striped-even--n-child: even;
```
Comment (srambach) [inline src/patternfly/components/Table/table.scss:308]: Are these used?

File: src/patternfly/components/Table/examples/Table.md [hunk 2]
```diff
@@ -3636,6 +3633,481 @@ For sticky columns to function correctly, the parent table's width must be contr
...
+      {{#> table-td table-td--data-label="Workspaces"}}
+        5
+      {{/table-td}}
+      {{#> table-td table-td--data-label="Last commit"}}
+        2 days ago
+      {{/table-td}}
+    {{/table-tr}}
+  {{/table-tbody}}
+
+  {{#> table-tbody table-tbody--modifier="pf-m-striped-even"}}
```
Comment (srambach) [inline src/patternfly/components/Table/examples/Table.md:3962]: Could the text in the cells be modified to be a little more obvious/explanatory that there are rowgroups causing the need for `pf-m-striped-even`?

**Review by srambach — COMMENTED** [review]
Just a couple of questions. Nice job.

**Review by mcarrano — APPROVED** [review]
Looks good to me!

**Review by mattnolting — APPROVED** [review]
L🍕 TM!

**Pull request merged**